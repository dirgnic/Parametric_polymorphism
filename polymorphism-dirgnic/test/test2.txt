def choose[A](cond : Bool, x : A, y : A) {
  if cond then x else y;
}

def main() {
  let z = choose[Int](true, 1, 2);
  return 0;
}
