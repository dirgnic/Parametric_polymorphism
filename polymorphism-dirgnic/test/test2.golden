=== After Monomorphization ===
Def "main" [] [] (Let "z" (App (App (App (Var "choose_TInt") (BoolLit True)) (IntLit 1)) (IntLit 2)) (IntLit 0))
Def "choose_TInt" [] [("cond",TBool),("x",TInt),("y",TInt)] (If (Var "cond") (Var "x") (Var "y"))
Num 0
